django/db/models/fields/related.py

```diff
@@ -1375,6 +1376,24 @@ def _check_relationship_model(self, from_model=None, **kwargs):
 
         return errors
 
+    def _check_table_uniqueness(self, **kwargs):
+        registered_tables = {
+            model._meta.db_table: model
+            for model in self.opts.apps.get_models(include_auto_created=True)
+            if model != self.remote_field.through
+        }
+        if not isinstance(self.remote_field.through, str) and self.m2m_db_table() in set(registered_tables.keys()):
```

Comment:
I added the isinstance check because test_missing_relationship_model in invalid_models_tests.test_relative_fields.RelativeFieldTests was failing with the following traceback:

```
Traceback (most recent call last):
  File "/Users/berkerpeksag/projects/django/tests/invalid_models_tests/test_relative_fields.py", line 299, in test_missing_relationship_model
    errors = field.check(from_model=Group)
  File "/Users/berkerpeksag/projects/django/django/db/models/fields/related.py", line 1143, in check
    errors.extend(self._check_table_uniqueness(**kwargs))
  File "/Users/berkerpeksag/projects/django/django/db/models/fields/related.py", line 1385, in _check_table_uniqueness
    print('m2m_db_table', self.m2m_db_table())
  File "/Users/berkerpeksag/projects/django/django/utils/functional.py", line 15, in _curried
    return _curried_func(*(args + moreargs), **dict(kwargs, **morekwargs))
  File "/Users/berkerpeksag/projects/django/django/db/models/fields/related.py", line 1476, in _get_m2m_db_table
    return self.remote_field.through._meta.db_table
AttributeError: 'str' object has no attribute '_meta'
```

Please let me know if there is a better way.